protecode/scanning.py

```diff
@@ -137,20 +138,24 @@ def iter_vulnerabilities_with_assessments(
                     continue
                 yield component, vulnerability, tuple(vulnerability.triages())
 
-        # only consider the most recent 50 products to import assessments from
-        # assumption: assessments are copied anyways from older assessments to
-        # newer ones so considering old assessments won't have any extra value
-        products_to_import_from.sort(
-            key=lambda p: p.product_id(),
-            reverse=True,
-        )
-        products_to_import_from = products_to_import_from[:50]
-
-        # do a shuffle to increase cache hits as otherwise same product ids are
-        # more likely to be handled simultaneously among the different threads
-        random.shuffle(products_to_import_from)
-
         for product in products_to_import_from:
+            if delete_inactive_products_after:
```

Comment:
I would structure this slightly different (also to maybe reduce nesting):

- calculate delete_after-timestamp outside of (before, obviously..) for-loop + fixate "now" outside of for-loop (rationale: if processing will take long, this way, there will be consistent (and even identical) timestamps; too my experience, this often has benefits)
- first add deletion-date (if not already present), then act upon deletion-date (if present); rationale: this way, passing-in of "delete after 0s" will work without special-case handling